contracts/src/v0.8/vrf/dev/SubscriptionAPI.sol

```diff
@@ -405,14 +410,16 @@ abstract contract SubscriptionAPI is ConfirmedOwner, IERC677Receiver, IVRFSubscr
     if (consumers.length == MAX_CONSUMERS) {
       revert TooManyConsumers();
     }
-    mapping(uint256 => uint64) storage nonces = s_consumers[consumer];
-    if (nonces[subId] != 0) {
+    ConsumerConfig storage consumerConfig = s_consumers[consumer][subId];
+    if (consumerConfig.active) {
       // Idempotence - do nothing if already added.
       // Ensures uniqueness in s_subscriptions[subId].consumers.
       return;
     }
-    // Initialize the nonce to 1, indicating the consumer is allocated.
-    nonces[subId] = 1;
+    // consumerConfig.nonce is 0 if the consumer had never sent a request to this subscription
```

Comment:
Should we consider starting nonce at 1, to maintain parity with previous implementation

Reply:
I think 0 makes more sense because previously 0 indicated inactive consumer but now, we have a `active` boolean flag. 

Also, we increment nonce before generating the first requestID [here](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/vrf/dev/VRFCoordinatorV2_5.sol#L296). So if we start the nonce at 1 here, first requestID will be generated with nonce 2

Reply:
Also, as long as the boolean is `true` to start the SSTORE will not be cold.